dh/sidh/internal/p434/arith_amd64.s

```diff
@@ -29,27 +29,36 @@
 //   |-128-| x |--- 256 ---| = |------ 384 ------|
 // Assuming the first digit multiplication was already performed.
 #define MULX128x256(I1, M1, T1, T2, T3, T4, T5)    \
-    MULXQ   M1+ 8(SB), T4, T2   \
+    MOVQ    M1+ 8(SB), AX       \
+    MULXQ   AX, T4, T2          \
     XORQ    AX, AX              \
```

Comment:
This XOR becomes useless because of the change afterwards (which seems unnecessary in itself).

Reply:
Although I might misunderstand the original code. In the original code AX is zeroed, then we use it to add OF to T4, and then zero it again?!  Either the code is not optimal or I'm missing something obvious.

Reply:
In addition to zeroing `AX`, the `XOR AX,AX` cleans both the carry and overflow bits. So we can start safely the ADCX and ADOX instructions.